ranker_trainer/LightGBMRankerTrainer/__init__.py

```diff
@@ -0,0 +1,71 @@
+import os
+from typing import Dict
+
+import lightgbm as lgb
+
+from jina.executors.rankers.trainer import RankerTrainer
+
+
+class LightGBMRankerTrainer(RankerTrainer):
+    """Ranker trainer to train the `LightGBMRanker` to enable offline/online-learning.
+
+    :param model_path: Path to the pretrained model previously trained using LightGBM.
+    :param param: Parameters for training.
+    :param train_set: Data to be trained on.
+    :param num_boost_round: Number of boosting iterations.
+    :param valid_sets: List of data to be evaluated on during training.
+    :param valid_names: Names of valid_sets.
+    """
+
+    def __init__(
+        self,
+        model_path: str,
+        train_set: lgb.Dataset,
+        param: Dict,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.model = None
+        self.param = param
+        self.train_set = train_set
+        self.model_path = model_path
+        self._is_trained = False
+
+    def post_init(self):
+        """Load the model."""
+        super().post_init()
+        if os.path.exists(self.model_path):
+            self.model = lgb.Booster(model_file=self.model_path)
+        else:
+            raise FileNotFoundError(f'The model path {self.model_path} not found.')
```

Comment:
If the model path does not exist, maybe is because u want to train from scratch, put a message to warn and ad `self.model = None`